main_test.go

```diff
@@ -87,8 +111,16 @@ func runWithConfig(t *testing.T, cfg *config.Config) (string, error) {
 	}
 
 	var sb strings.Builder
+	args := []string{"--config", configPath}
 
-	err := run(&sb, []string{"--config", configPath})
+	if len(sarifReportPath) > 0 {
+		args = append(args, "--sarif-report-path", sarifReportPath[0])
+	}
+	err := run(&sb, args)
+	if len(sarifReportPath) > 0 {
+		file, _ := os.ReadFile(sarifReportPath[0])
+		return string(file), err
```

Comment:
Let's not mix these two things here. `runWithConfig` returns the output of the program and shouldn't conditionally return the sarif file output. It also silently swallows the `err` if the sarif report was not created.

If you need to read back the created sarif file in a test, just do `file, err := os.ReadFile(sarifReportPath)` there explicitly. This will also make the testcase clearer to understand.

Reply:
You can also create test helpers to compare two files.

Example:

```go
func assertFileContentsEqual(t *testing.T, actualPath string, expectedPath string) {
  actualContents := mustReadFile(t, actualPath)
  expectedContents := mustReadFile(t, expectedPath)
  assert.Equal(t, string(expectedContents), string(actualContents))
}

func mustReadFile(t *testing.T, path string) []byte {
  contents, err := os.ReadFile(path)
  require.NoErrorf(t, err, "failed to load file %q", path)
  return contents
}
```